fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocks.g4

```diff
@@ -1759,7 +1759,7 @@ functionCall
     | specialFunctionExpression                                                           #specialFunction
     | aggregationFunction over?                                                           #aggregationFunctionCall
     | windowFunction over                                                                 #windowFunctionCall
-    | qualifiedName '(' (expression (',' expression)*)? ')'  over?                        #simpleFunctionCall
+    | qualifiedName '(' DISTINCT? (expression (',' expression)*)? ')'  over?              #simpleFunctionCall
```

Comment:
I get your idea, the idea is right, but you can't modify as that.

rewrite array_agg(distinct x) -> array_distinct(array_agg(x)):
1. array_agg aggregate function add distinct check and analye to aggregate function
2. rewrete  array_agg(distinct x) -> array_distinct(array_agg(x)) in logical rewrite phase

you don't need modify syntax file

Reply:
IIUC, if we donot add `DISTINCT` in syntax file, it will be failed in SQL parse phase.